api/v1beta1/clusterclass_types.go

```diff
@@ -518,7 +521,125 @@ type JSONSchemaProps struct {
 	// NOTE: Can be set for all types.
 	// +optional
 	Default *apiextensionsv1.JSON `json:"default,omitempty"`
-}
+
+	// XValidations describes a list of validation rules written in the CEL expression language.
+	// +optional
+	XValidations ValidationRules `json:"x-kubernetes-validations,omitempty"`
+}
+
+// ValidationRules describes a list of validation rules written in the CEL expression language.
+// +listType=map
+// +listMapKey=rule
+type ValidationRules []ValidationRule
```

Comment:
What's the advantage of defining a type alias here? I would probably have just gone inline for this one unless there's a specific validation you want to add that cannot be done without a type alias

Reply:
Not sure. One reason could be to keep it in sync with upstream, but I think we don't have to go that far.

Is the type alias necessary for the listType/listMapKey because controller-gen doesn't support it otherwise?

If not. Let's drop ValidationRules